vsdownload.py

```diff
@@ -88,12 +97,34 @@ def setPackageSelectionMSVC15(args, packages, userversion, sdk, toolversion, def
         args.package.extend(defaultPackages)
 
 def setPackageSelection(args, packages):
+    if not args.architecture:
+        if args.host_arch is not None:
+            args.architecture = [ args.host_arch ]
+        else:
+            args.architecture = ["x86", "x64", "arm", "arm64"]
+
     # If no packages are selected, install these versionless packages, which
     # gives the latest/recommended version for the current manifest.
-    extraarchs = ["ARM", "ARM64"]
-    defaultPackages = ["Microsoft.VisualStudio.Workload.VCTools", "Microsoft.VisualStudio.Component.VC.ATL"]
-    for arch in extraarchs:
-        defaultPackages.extend(["Microsoft.VisualStudio.Component.VC.Tools." + arch, "Microsoft.VisualStudio.Component.VC.ATL." + arch])
+    defaultPackages = []
+    if "x86" in args.architecture or "x64" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
+        defaultPackages.append("Microsoft.VisualCpp.ASAN.X86")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL")
+    if "arm" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.ARM")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL.ARM")
+    if "arm64" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.ARM64")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL.ARM64")
+    
+    # TODO
+    defaultPackages.append("Win11SDK_10.0.22621")
+
+    defaultPackages.extend([
```

Comment:
I presume you're adding these packages manually here, to compensate for things that are left out by installing `Microsoft.VisualStudio.Component.VC.Tools.x86.x64` rather than `Microsoft.VisualStudio.Workload.VCTools`?